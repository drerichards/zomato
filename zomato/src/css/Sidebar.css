.Sidebar {
    flex: 2 2 15%;
    padding: 0 1%;
}

.Sidebar form {
    padding: 10% 5%;
}

.Sidebar form p {
    border-bottom: 0.5px solid rgba(238, 238, 238, 0.842);
}

.Sidebar [type="checkbox"]:checked+label:before {
    border-right: 2px solid #F25F5C;
    border-bottom: 2px solid #F25F5C;
}