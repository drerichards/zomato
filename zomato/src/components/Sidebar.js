import React from 'react'
import '../css/Sidebar.css'

const Sidebar = () => {
    const onCuisineClick = (e) => {
        e.currentTarget.classList.contains('checked') ?
            e.currentTarget.classList.remove('checked')
            : e.currentTarget.classList.add('checked')
    }
    return (
        <div className='Sidebar'>
            <form action="#" className='card'>
                <p>
                    <input type="checkbox" id="1" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="1">American</label>
                </p>
                <p>
                    <input type="checkbox" id="152" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="152">African</label>
                </p>
                <p>
                    <input type="checkbox" id="3" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="3">Asian</label>
                </p>
                <p>
                    <input type="checkbox" id="193" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="193">BBQ</label>
                </p>
                <p>
                    <input type="checkbox" id="158" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="158">Caribbean</label>
                </p>
                <p>
                    <input type="checkbox" id="40" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="40">Fast Food</label>
                </p>
                <p>
                    <input type="checkbox" id="143" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="143">Healthy Food</label>
                </p>
                <p>
                    <input type="checkbox" id="148" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="148">Indian</label>
                </p>
                <p>
                    <input type="checkbox" id="55" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="55">Italian</label>
                </p>
                <p>
                    <input type="checkbox" id="136" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="136">Latin American</label>
                </p>
                <p>
                    <input type="checkbox" id="137" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="137">Middle Eastern</label>
                </p>
                <p>
                    <input type="checkbox" id="83" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="83">Seafood</label>
                </p>
                <p>
                    <input type="checkbox" id="461" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="461">Soul Food</label>
                </p>
                <p>
                    <input type="checkbox" id="95" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="95">Thai</label>
                </p>
                <p>
                    <input type="checkbox" id="308" onClick={(e) => { onCuisineClick(e) }} />
                    <label htmlFor="308">Vegetarian</label>
                </p>
            </form>
        </div>
    )
}

export default Sidebar