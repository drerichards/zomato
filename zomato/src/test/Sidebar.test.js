import React from 'react'
import renderer from 'react-test-renderer'
import Sidebar from '../components/Sidebar'

test('Sidebar snapshot test', () => {
    const component = renderer.create(<Sidebar />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
