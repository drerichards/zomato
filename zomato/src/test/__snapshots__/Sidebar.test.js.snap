// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar snapshot test 1`] = `
<div
  className="Sidebar"
>
  <form
    action="#"
    className="card"
  >
    <p>
      <input
        id="1"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="1"
      >
        American
      </label>
    </p>
    <p>
      <input
        id="152"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="152"
      >
        African
      </label>
    </p>
    <p>
      <input
        id="3"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="3"
      >
        Asian
      </label>
    </p>
    <p>
      <input
        id="193"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="193"
      >
        BBQ
      </label>
    </p>
    <p>
      <input
        id="158"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="158"
      >
        Caribbean
      </label>
    </p>
    <p>
      <input
        id="40"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="40"
      >
        Fast Food
      </label>
    </p>
    <p>
      <input
        id="143"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="143"
      >
        Healthy Food
      </label>
    </p>
    <p>
      <input
        id="148"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="148"
      >
        Indian
      </label>
    </p>
    <p>
      <input
        id="55"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="55"
      >
        Italian
      </label>
    </p>
    <p>
      <input
        id="136"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="136"
      >
        Latin American
      </label>
    </p>
    <p>
      <input
        id="137"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="137"
      >
        Middle Eastern
      </label>
    </p>
    <p>
      <input
        id="83"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="83"
      >
        Seafood
      </label>
    </p>
    <p>
      <input
        id="461"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="461"
      >
        Soul Food
      </label>
    </p>
    <p>
      <input
        id="95"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="95"
      >
        Thai
      </label>
    </p>
    <p>
      <input
        id="308"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="308"
      >
        Vegetarian
      </label>
    </p>
  </form>
</div>
`;
